lib/tools/system-calls.js

```diff
@@ -922,6 +921,10 @@ systemCallMethods.changeUserPrivileges = async function changeUserPrivileges (is
  * @return {rootResult}
  */
 systemCallMethods.root = async function root () {
+  // If it's already rooted, our job is done. No need to root it again.
+  if (await this.isRoot()) {
```

Comment:
I assume this call should be inside `changeUserPrivilege` method. So, if isRoot is true and isElevated true or !isRoot and !isElevated we should do a short circuit. Also, this will fix the initial value of wasAlreadyRooted

Reply:
I could do that... I"ll have to pick it up again tomorrow, because when I tried that way a broke a bunch of unit tests.

Reply:
Never mind... I fixed it now. It was easy.